### PR TITLE
Adds unit tests, e2e tests, manual tests, and travis.yml

### PR DESCRIPTION
**Pull request by benjaminapetersen**
- Move /demo/ to /test/manual/ as these are manual test files
- gulp test-e2e will use protractor + the /test/manual files to automate manual tests
- gulp test-unit will run unit tests
- gulp tdd will run the unit tests while watching all files for changes (repeat tests)
- Fix delete-link change color when sort disabled, add classes for validation help text
- Add classes to inputs & links
- jshint updates
- Add unit test for provider
- Update README.md
- Add validation tests, add validate.sh and travis.yml
- add _tmp-css task for validate.sh to validate build

@jwforres @rhamilto

### CODE REVIEW

**Comment by benjaminapetersen** [comment]
Ooooo there goes Travis.  Is just a diff check though, will see if I can run tests as well.

**Comment by benjaminapetersen** [comment]
[TEST]

**Comment by benjaminapetersen** [comment]
Travis is setup, working on getting build validation verified & unit tests run

**Comment by benjaminapetersen** [comment]
[TEST]

**Comment by jwforres** [comment]
we dont have jenkins hooked up to this repo so i dont think [TEST] is going to do anything...

**Comment by jwforres** [comment]
travis will run again automatically any time you push a new commit ID, so you can trigger it again by amending your commit and force pushing to the branch

OR  to run again on the same commit just go to the travis UI and there is a rebuild button

**Comment by benjaminapetersen** [comment]
ah oh neat. :) was curious bout [TEST] but ive been tweaking the commit a bit anyway.

**Comment by benjaminapetersen** [comment]
@jwforres got `validate-dist` running well, but `phantomJS` doesn't run the unit tests the same way, so a couple of them fail.  I'm inclined to merge this PR anyway, unit tests run locally. thoughts @jwforres?

**Comment by benjaminapetersen** [comment]
@jwforres  validate dist runs, unit tests run.  Interestingly, e2e style tests with a real browser are probably the most valuable for this thing since it relies on user input so much.  Will just have to manually run them whenever updating (and probably add more, but this is a pretty decent sized start).

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#cache:
+#  directories:
+#  - node_modules
+#  - bower_components
+before_install:
+  - "export DISPLAY=:99.0"
+  - "sh -e /etc/init.d/xvfb start"
+  - sleep 3 # give xvfb some time to start
+before_script:
+# apt-get install xvfb
```
Comment (jwforres) [inline .travis.yml]: leave these commented out for a reason?

Comment (benjaminapetersen) [inline .travis.yml]: Ah, nope, I can remove that.

Comment (benjaminapetersen) [inline .travis.yml]: Actually, quick Q.  Should I uncomment the `cache` blocks?  I believe we do in the web console to speed things up. Not sure if its valuable here.

Comment (benjaminapetersen) [inline .travis.yml]: Removed it, can add later if caching is useful.  Will merge this soon as the tests pass to get it off the books.

**Pull request merged**